### DISCUSSION TITLE
Tuta wrongly claims that certain valid e-mail addresses are invalid

### DISCUSSION THREAD

**Discussion by realpixelcode** [tutao/tutanota#8626] [General]
- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

**Describe the bug**

When entering certain recipient e-mail addresses that are valid according to [RFC 5322](https://www.rfc-editor.org/rfc/rfc5322), Tuta wrongly complains that they're allegedly invalid.

**To Reproduce**

1. Draft a new e-mail.
2. Enter any of the examples listed below as the recipient's address.
3. Send the e-mail.
4. Error: “Please correct the invalid addresses: …”

[According to Wikipedia](https://en.wikipedia.org/wiki/Email_address#Valid_email_addresses), all of the following e-mail addresses are valid, but Tuta accepts none of them:

- john.smith(comment)@example.com
- (comment)john.smith@example.com
- john.smith@(comment)example.com
- john.smith@example.com(comment)
- " "@test.example.org
- "very.(),:;<>[]\".VERY.\"very@\\ \"very\".unusual"@strange.example.com
- postmaster@[123.123.123.123]
- postmaster@[IPv6:2001:0db8:85a3:0000:0000:8a2e:0370:7334]
- _test@[IPv6:2001:0db8:85a3:0000:0000:8a2e:0370:7334]

Notably, when entering `" "@example.org`, Tuta misinterprets it as `"@example.org` but still successfully sends the e-mail. When using a sub-domain, however, such as `" "@test.example.org`, Tuta also attempts to send the e-mail but then sends back an error message that reads:

> We're sorry but the following e-mail could not be delivered.
> Date: Dec 9, 2024, 8:57:32 PM
> Recipients: "@test.example.org
> 501 5.1.3 Bad recipient address syntax

**Expected behavior**

Tuta should accept all valid e-mail addresses without complaining.

**Screenshots**

![grafik](https://github.com/user-attachments/assets/1c9bf181-2f79-4c5c-8999-817179da7ec1)

![grafik](https://github.com/user-attachments/assets/2c92bada-45f5-4e20-bdec-5925e4787f63)

![grafik](https://github.com/user-attachments/assets/e63ebe72-275c-4b68-a241-aa701195dc9d)

**Desktop (please complete the following information):**
 - Desktop Environment, if applicable (on Linux)

**Additional context**
 
none

**Comment by hrb-hub** [comment]
Hi @realpixelcode , thank you for the report and effort you've put into this!
While your point is valid, I'm afraid that there are no current plans to address this issue, as the provided examples are not supported by most email providers. However that is not to say that it won't be addressed in the future.

Closing the issue for now, thanks again :)

**Comment by realpixelcode** [comment]
While I do agree that e-mail addresses that contain other symbols than just alphanumeric characters, dots and underscores are not particularly _common_ among regular users, I think it is a fundamentally wrong choice not to make Tuta **_compatible_** with **_the e-mail standard_**.

Again, this is **_not_** about letting users **_create_** weird addresses on tuta.com¹; it is about allowing users to **_send_** e-mails to **_all valid addresses_**.

> the provided examples are not supported by most email providers

By far not every e-mail provider disallows **_sending_** e-mails to uncommon but valid addresses. And even if almost everyone else did, would that really be a valid excuse for Tuta to do the same? Most providers don't care about privacy either – and yet, Tuta uses E2EE, for example.

There are also a couple good use cases for having an uncommon e-mail address, which means there are valid reasons to send e-mails to such addresses:

- internal corporate notification systems or mailing lists
- avoiding spam
- being able to attribute e-mails to specific senders
- having a cool-looking address

¹ (though I would find it great if I could create arbitrary e-mail addresses on my own domain without self-hosting a mail server)

**Comment by hrb-hub** [comment]
Hi @realpixelcode ,
Your points are valid, there's no disagreement there. What I meant by
> the provided examples are not supported by most email providers

is that while complying with the spec and supporting all valid email addresses is the correct thing to do, it isn't something that most users care about. And because there are quite a few things that users do want and ask for regularly, those take precedence.
I hope this clarifies things :)
